src/lib/components/button/main-button.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+  const getButtonState = (disabled?: boolean) => {
+    if (disabled) {
+      return 'disabled';
+    }
+    return 'default';
+  };
+
+  const getButtonStyle = (variant) => {
+    switch (variant) {
+      case 'secondary':
+        return 'secondary';
+      case 'destroy':
+        return 'destroy';
+      case 'primary':
+      default:
+        return 'primary';
+    }
+  };
+
+  const getButtonSize = (size) => {
+    switch (size) {
+      case 'small':
+        return 'small';
+      case 'default':
+      default:
+        return 'default';
+    }
+  };
+
+  export let disabled: boolean = false;
+  export let variant: string = '';
+  export let active: boolean = false;
+  export let size: string = 'default';
```

Comment:
The same goes here as well.

Reply:
since all of these don't need to be set on the button "default" settings is there another way around this?